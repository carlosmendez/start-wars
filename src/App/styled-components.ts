import styled from '@emotion/styled';

export const StyledMain = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
`;
