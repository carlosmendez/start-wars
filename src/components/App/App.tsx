import React, { ReactNode } from 'react';
import { Global, css } from '@emotion/react';
import emotionNormalize from 'emotion-normalize';
import { StyledMain } from "./styled-components";

type AppProps = {
  children: ReactNode;
};

const GlobalStyles = () => <Global styles={css`
  ${emotionNormalize}
  body {
    color: #feda4a;
    background-color: #000;
    font-family: sans-serif;
  }
`}
/>;

const App = ({
 children,
}: AppProps) => {
  return (
    <>
      <GlobalStyles/>
      <StyledMain>{children}</StyledMain>
    </>
  )
}

export { App };
