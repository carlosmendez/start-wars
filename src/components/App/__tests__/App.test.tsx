import React from 'react';
import renderer from 'react-test-renderer';
import { App } from "../App";

const MockChildren = () => <div>....mockChildrenComponent</div>;

describe('App component', () => {
  it('renders correctly', () => {
    const jsx = (
      <App>
        <MockChildren />
      </App>
    );
    const tree = renderer
      .create(jsx)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
