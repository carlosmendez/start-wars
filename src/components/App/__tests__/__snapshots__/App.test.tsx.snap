// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<main
  className="css-1in8n8a"
>
  <div>
    ....mockChildrenComponent
  </div>
</main>
`;
